test/documentation_test.rb

```diff
@@ -0,0 +1,140 @@
+        </div>
+        <input type="submit" name="commit" value="Log In" class="btn btn-secondary" data-disable-with="Log In" />
+      </form>
+      HTML
+
+      actual = bootstrap_form_with(model: @user, local: true) do |f|
+        f.email_field(:email)
+         .concat(f.password_field(:password))
+         .concat(f.check_box(:remember_me))
+         .concat(f.submit("Log In"))
+      end
+
+      assert_equivalent_xml remove_default_ids_for_rails_5_1(expected), actual
+    end
+
+    test "form_with no model example" do
+      expected = <<-HTML.strip_heredoc
+      <form role="form" action="/subscribe" accept-charset="UTF-8" method="post">
+        <input name="utf8" type="hidden" value="&#x2713;" />
+        <div class="form-group">
+          <label for="email">Email</label>
+          <input value="name@example.com" class="form-control" type="email" name="email" />
+        </div>
+        <input type="submit" name="commit" value="Save " class="btn btn-secondary" data-disable-with="Save " />
+      </form>
+      HTML
+
+      actual = bootstrap_form_with(url: '/subscribe', local: true) do |f|
+        f.email_field(:email, value: 'name@example.com')
+         .concat(f.submit)
+      end
+
+      assert_equivalent_xml expected, actual
+    end
+
+    test "form_with explicit id example" do
+      expected = <<-HTML.strip_heredoc
+      <form role="form" action="/users" accept-charset="UTF-8" method="post">
+        <input name="utf8" type="hidden" value="&#x2713;" />
+        <div class="form-group">
+          <label class="required" for="user_email">Email</label>
+          <input class="form-control" type="email" value="steve@example.com" name="user[email]" />
+        </div>
+        <div class="form-group">
+          <label for="password">Password</label>
+          <input id="password" class="form-control" type="password" name="user[password]" />
+          <small class="form-text text-muted">A good password should be at least six characters long</small>
+        </div>
+        <input type="submit" name="commit" value="Log In" class="btn btn-secondary" data-disable-with="Log In" />
+      </form>
+      HTML
+
+      # Removed this from above pending #357
+      # <div class="form-check">
+      #   <label for="remember" class="form-check-label">
+      #   <input name="user[remember_me]" type="hidden" value="0" />
+      #   <input id="remember" class="form-check-input" type="checkbox" value="1" name="user[remember_me]" /> Remember me</label>
+      # </div>
+
+      actual = bootstrap_form_with(model: @user, local: true) do |f|
+        f.email_field(:email)
+         .concat(f.password_field(:password, id: :password))
+         .concat(f.submit("Log In"))
+      end
+      #  .concat(f.check_box(:remember_me, id: :remember)) pending #357 place above "Log In"
+
+      assert_equivalent_xml expected, actual
+    end
+  end
+end
```

Comment:
Why not just have tests that test something specific instead of shotgun/integration approach here? Don't think this entire test file is needed.

Reply:
The intent of this file to test that what we're putting in the documentation, is in fact what the helpers produce.

Reply:
I see what you are going for with this test. As Bootstrap and Rails change, the examples in the README will become out of date, which potentially misleads or confusion people learning about the gem. It also reflects poorly on the project if the README is inaccurate.

However I'm not convinced this is the best solution. I'd like to remove this from this PR and save it for future consideration.